tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -670,6 +670,8 @@ private void OnHostingHttpRequestInStop(object arg)
                 AspNetCoreRequestHandler.StopAspNetCorePipelineScope(tracer, CurrentSecurity, rootScope, httpContext);
             }
 
+            CoreHttpContextStore.Instance.Remove();
```

Comment:
I wonder if you should call this in `AspNetCoreRequestHandler.StopAspNetCorePipelineScope()` as well 🤔 `StopAspNetCorePipelineScope` is called outside of this pipeline too, in Azure Functions for example, so may be safest to add a call there (calling it twice should be safe I think?)

Reply:
good point!! thanks 
[4d5bb45](https://github.com/DataDog/dd-trace-dotnet/pull/6151/commits/4d5bb450ce1ae264a4ee7ff0a8902d60f970f108)
yes it should be safe as its just setting a value to null..